src/animation_player.rs

```diff
@@ -127,69 +159,74 @@ pub fn match_bones(
         let current_frame = FrameData(current_frame);
         let next_frame = FrameData(next_frame);
 
-        for (entity, name, mut transform, global_transform) in q_names.iter_mut() {
-            let bone_name = &name.as_str()[6..];
-
-            let mut joint_index: usize = 0;
-
-            for joint in bvh_animation_data.joints() {
-                if bone_name == joint.data().name() {
-                    commands.entity(entity).insert(BoneIndex(joint_index));
-
-                    let offset = joint.data().offset();
-
-                    let mut current_translation = Vec3::new(offset.x, offset.y, offset.z);
-                    let mut next_translation = Vec3::new(offset.x, offset.y, offset.z);
-
-                    let channels = joint.data().channels();
-
-                    let current_rotation;
-                    let next_rotation;
-
-                    if channels.len() == 3 {
-                        current_rotation = current_frame.get_rotation(channels);
-                        next_rotation = next_frame.get_rotation(channels);
-                    } else {
-                        let current_offset;
-                        let next_offset;
-                        (current_offset, current_rotation) =
-                            current_frame.get_translation_rotation(channels);
-                        (next_offset, next_rotation) =
-                            next_frame.get_translation_rotation(channels);
-
-                        current_translation += current_offset;
-                        next_translation += next_offset;
-                    }
-
-                    let interpolated_translation =
-                        Vec3::lerp(current_translation, next_translation, interpolation_factor);
-
-                    let interpolated_rotation =
-                        Quat::slerp(current_rotation, next_rotation, interpolation_factor);
+        let mut joint_index: usize = 0;
+
+        for joint in bvh_animation_data.joints() {
+            let bone_names = joint.data().name().to_str().unwrap();
+            for bone_map in q_bone_map.iter() {
+                if let Some(bone_entity) = &bone_map.0.get(bone_names) {
+                    commands
+                        .entity(**bone_entity)
+                        .insert(BoneIndex(joint_index));
```

Comment:
`Entity` implements `Copy` so we can do sth like this without cloning.
```suggestion
                if let Some(&bone_entity) = bone_map.0.get(bone_names) {
                    commands
                        .entity(bone_entity)
                        .insert(BoneIndex(joint_index));
```